pkg/obs/controllerserver.go

```diff
@@ -29,7 +32,51 @@ type controllerServer struct {
 
 func (cs *controllerServer) CreateVolume(_ context.Context, req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	log.Infof("CreateVolume: called with args %v", protosanitizer.StripSecrets(*req))
+	credentials := cs.Driver.cloud
+
+	volName := req.GetName()
+	if err := createVolumeValidation(volName, req.GetVolumeCapabilities()); err != nil {
+		return nil, err
+	}
+
+	if vol, err := services.GetBucket(credentials, volName); err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	} else if vol != "" {
+		return buildCreateVolumeResponse(volName), nil
+	}
+
+	_, err := services.CreateBucketWithTag(credentials, volName)
```

Comment:
It is recommended that `CreateBucketWithTag` be split into `CreateBucket` and `CreateBucketTag`.

Reply:
down